development/workflow.rst

```diff
@@ -0,0 +1,305 @@
+
+Ensure that Git is setup to use your *institution-hosted* email address the :file:`~/.gitconfig` file.
+You can do this from the command line:
+
+.. code-block:: bash
+   git config --global user.name "Your Name"
+   git config --global user.email "your_email@institution.edu"
+
+Likewise, in your `GitHub account email settings <https://github.com/settings/emails>`_, add your institution-hosted email (e.g. @lsst.org).
+Set this institutional email as your **Primary GitHub** email address.
+
+These steps are important for source code and copyright audits from Git commit history.
+
+See :ref:`git-setup-best-practices` for best practices in configuring Git and GitHub.
+
+.. _git-branching:
+
+DM Git Branching Policy
+=======================
+
+Rather than forking LSST's GitHub repositories, DM developers should work with clones of the repositories in LSST's GitHub organizations (`lsst <https://github.com/lsst>`_, `lsst-dm <https://github.com/lsst>`_, and `lsst-sqre <https://github.com/lsst>`_).
+Since the GitHub ``origin`` remotes are shared, it is essential that DM developers adhere to the following naming conventions for branches.
+
+The master branch
+-----------------
+
+``master`` is the main integration branch for our repositories.
+Development is not done directly on the ``master`` branch, but instead on *ticket branches*.
+*Trivial* documentation fixes are the only scenarios where working directly on ``master`` and by-passing the code review process is permitted.
+
+The Git history of ``master`` should never be re-written with force pushes.
+
+User branches
+-------------
+
+You can do experimental, proof-of-concept work in 'user branches.'
+
+These branches are named
+
+.. code-block:: text
+
+   u/{{username}}/{{topic}}
+
+User branches can be pushed to GitHub to enable collaboration and communication.
+Colleagues should be cognizant that user branches are for prototyping before offering unsolicited code reviews.
+
+Developers can feel free to rebase and force push work to their personal user branches.
+
+A user branch *cannot* be merged into master; it must be converted into a *ticket branch* first.
+
+Ticket branches
+---------------
+
+Ticket branches are associated with a JIRA ticket.
+Only ticket branches can be merged into ``master``.
+
+If the JIRA ticket is named ``DM-NNNN``, then the ticket branch will be named
+
+.. code-block:: text
+
+   tickets/DM-NNNN
+
+A ticket branch can be made by branching off an existing user branch.
+This is a great way to formalize experimental work and shaping it into an LSST software contribution.
+
+When code on a ticket branch is ready for review and merging, follow the :ref:`code review process documentation <git-code-review>`.
+
+.. _git-code-review:
+
+DM Code Review and Merging Process
+==================================
+
+When development on your ticket branch is complete, there is a standard process for reviewing your work and merging it into the ``master`` integration branch.
```

Comment:
Cribbing text from the CodeReviewProcess would seem appropriate to answer that question.